src/basic/env-util.c

```diff
@@ -779,7 +779,7 @@ int serialize_environment(FILE *f, char **environment) {
                 if (!ce)
                         return -ENOMEM;
 
-                fprintf(f, "env=%s\n", *e);
+                fprintf(f, "env=%s\n", ce);
```

Comment:
Talking of some magic syntax, I think it would we great if the output of `show-environment` could be used as the value of `DefaultEnvironment` and could be passed to `set-environment`. It's not possible now. Not that it matters though :-)